modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalRequestIT.java

```diff
@@ -121,6 +125,18 @@ public void testPrecisionAtRequest() {
                 }
             }
         }
+
+        // test that a different window size k affects the result
+        metric = new PrecisionAtK(1, false, 3);
+        task = new RankEvalSpec(specifications, metric);
+        task.addIndices(indices);
+
+        builder = new RankEvalRequestBuilder(client(), RankEvalAction.INSTANCE, new RankEvalRequest());
+        builder.setRankEvalSpec(task);
+
+        response = client().execute(RankEvalAction.INSTANCE, builder.request()).actionGet();
+        expectedPrecision = (1.0 / 3.0 + 2.0 / 3.0) / 2.0;
+        assertEquals(0.5, response.getEvaluationResult(), Double.MIN_VALUE);
```

Comment:
do we want to put here `expectedPrecision` value instead of `0.5`?

Reply:
Good catch, and yes, I changed that locally on the branch already while I continued working on adding this optional parameter to MeanReciprocalRank and DCG as well.